app/upgrades/v4/upgrades.go

```diff
@@ -0,0 +1,68 @@
+package v4
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	stakingkeeper "github.com/cosmos/cosmos-sdk/x/staking/keeper"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+	upgradetypes "github.com/cosmos/cosmos-sdk/x/upgrade/types"
+)
+
+func CreateUpgradeHandler(
+	mm *module.Manager,
+	configurator module.Configurator,
+	sk stakingkeeper.Keeper,
+) upgradetypes.UpgradeHandler {
+	return func(ctx sdk.Context, plan upgradetypes.Plan, vm module.VersionMap) (module.VersionMap, error) {
+		ctx.Logger().Info("Starting upgrade for multi staking...")
+		fixMinCommisionRate(ctx, sk)
+		return mm.RunMigrations(ctx, configurator, vm)
+	}
+}
+
+func fixMinCommisionRate(ctx sdk.Context, staking stakingkeeper.Keeper) {
+	// Upgrade every validators min-commission rate
+	validators := staking.GetAllValidators(ctx)
+	newComm := sdk.MustNewDecFromStr(NewMinCommisionRate)
+	params := staking.GetParams(ctx)
+	params.MinCommissionRate = newComm
+	staking.SetParams(ctx, params)
+	for _, v := range validators {
+		//nolint
+		if v.Commission.Rate.LT(newComm) {
+			// We need to remove
+			comm, err := updateValidatorCommission(ctx, staking, v, newComm)
+			if err != nil {
+				panic(err)
+			}
+
+			v.Commission = comm
+
+			// call the before-modification hook since we're about to update the commission
+			staking.BeforeValidatorModified(ctx, v.GetOperator())
+			staking.SetValidator(ctx, v)
```

Comment:
is there staking.AfterValidatorModified() ?

Reply:
No, it just have AfterValidatorCreated, AfterValidatorRemoved, AfterValidatorBonded and AfterValidatorBeginUnbonding